src/rapids_singlecell/preprocessing/_scale.py

```diff
@@ -245,12 +265,191 @@ def _scale_sparse_csr(
         scale_csr(
             (X.shape[0],),
             (64,),
-            (X.indptr, X.indices, X.data, std, mask_array, max_value, X.shape[0]),
+            (
+                X.indptr,
+                X.indices,
+                X.data,
+                std.astype(X.dtype),
+                mask_array,
+                max_value,
+                X.shape[0],
+            ),
         )
 
         return X, mean, std
 
 
+def _scale_dask(X, *, mask_obs=None, zero_center=True, inplace=True, max_value=None):
+    import dask
+
+    if not inplace:
+        X = X.copy()
+    if mask_obs is None:
+        mean, var = _get_mean_var(X)
+        mask_array = cp.ones(X.shape[0], dtype=cp.int32)
+
+    else:
+        mean, var = _get_mean_var(X[mask_obs, :])
+        mask_array = cp.array(mask_obs).astype(cp.int32)
+    mean, var = dask.compute(mean, var)
+    std = cp.sqrt(var)
+    std[std == 0] = 1
+    max_value = _get_max_value(max_value, X.dtype)
+
+    mask_array = da.from_array(
+        mask_array, chunks=(X.chunks[0],), meta=_meta_dense(mask_array.dtype)
+    )
+
+    if isinstance(X._meta, sparse.csr_matrix) and zero_center:
+        from ._kernels._sparse2dense import _sparse2densekernel
+
+        kernel = _sparse2densekernel(X.dtype)
+        kernel.compile()
+
+        def __dense(X_part):
+            major, minor = X_part.shape
+            dense = cp.zeros(X_part.shape, order="C", dtype=X_part.dtype)
+            max_nnz = cp.diff(X_part.indptr).max()
+            tpb = (32, 32)
+            bpg_x = math.ceil(major / tpb[0])
```

Comment:
tpb? bpg? please define in comment

Reply:
that are normal names for threads and blocks